### PR TITLE
metrics - upgrade py2 -> py3, #13164

### PR DESCRIPTION
**Pull request by clarketm**
This PR is the *first* of several for the incremental migration from pyhton2 -> python3, #13164. This migration will rollout project-by-project with this first targeting the `metrics` top-level directory (as the branch name implies). 

#### Notes:
1. One caveat to this incremental approach is interdependencies between projects (i.e. directories) which will be handled by PRing interdependent projects together. 

2. Another caveat is regarding linting; pylint does not have a version supporting both python2 and python3. Additionally, the **[bazelbuild/rules_python](https://github.com/bazelbuild/rules_python)** [pip_import](https://github.com/bazelbuild/rules_python#importing-pip-dependencies) directive uses the system python interpreter and thus cannot natively support dual install of python dependencies. For this reason, the pylint migration will occur last along with the pylint dependency upgrade (and any other subsequent dependency upgrades).

3. [gubernator](https://github.com/kubernetes/test-infra/tree/master/gubernator) will be migrated as a second effort as it is a standalone application with a sever runtime, and it has its own scoped dependencies which may require more attention.

### CODE REVIEW

File: py.bzl [hunk 1]
```diff
@@ -22,21 +22,6 @@ load(
 )
 
 def python_repos():
-    http_archive_with_pkg_path(
-        name = "ruamel_yaml",
```
Comment (cjwagner) [inline py.bzl]: Was this unused?

File: hack/verify-pylint.sh [hunk 2]
```diff
@@ -32,7 +33,9 @@ else
...
 
 export PYLINTHOME=$TEST_TMPDIR
-pylint="$(dirname $0)/pylint_bin"
 
 shopt -s extglob globstar
-${pylint} !(gubernator|external|vendor|bazel-*)/**/*.py
+
+# TODO(clarketm) there is no version of `pylint` that supports "both" PY2 and PY3
+# I am disabling pylint checks for python3 files until migration complete
+$"$DIR/pylint_bin" !(metrics|gubernator|external|vendor|bazel-*)/**/*.py
```
Comment (cjwagner) [inline hack/verify-pylint.sh]: Is the first `$` a typo?

File: py.bzl (hunk 1 above)
Comment (clarketm) [reply py.bzl]: @cjwagner - yes, this is dead, superseded code. I overlooked it before but and removing now after investigation of the bazel build protocol and the fact that `ruamel.yaml` is declared in `requirements.txt` and installed via `pip`.

File: hack/verify-pylint.sh (hunk 2 above)
Comment (clarketm) [reply hack/verify-pylint.sh]: @cjwagner - for some reason I thought this was required by bash for command substitution, but I was mistaken and will fix.

Comment (clarketm) [reply hack/verify-pylint.sh]: @cjwagner - [updated](89bdf78b1b7ffa5c27cfc19f8bc6a1d28b184f04)

**Pull request merged**